### DISCUSSION TITLE
Node ID

### DISCUSSION THREAD

**Discussion by arresse** [freeplane/freeplane#517] [General]
What's the fastest way to locate and copy a node ID # to the clipboard?

Thank you.

**Comment by euu2021** [comment]
You can set a hotkey to `Edit->Copy->Copy node ID`. To assign a hotkey use `Tools->Assign hot key`.

Maybe, it´s possible to use a listener in a script so the node ID is copied as soon as the node is selected. This would be the fastest way, but I'm not sure if it's possible. See [these scripts](https://sourceforge.net/p/freeplane/discussion/758437/thread/e9200db6c3/) by macmarrun as examples of the use of listeners.

**Reply by arresse** [reply]
Thank you eu2021. I should have seen the copy node id in the menu, thanks for pointing it out. Hotkey is set. I'm mostly using this to create internal links, so I have to add the # sign in front what is copied, but this is quicker than what I was doing.

Unfortunately, the listener scripts are a bit over my head right now. I'd love to learn the scripting business but don't have the time now. This will be my next FP project. Could you suggest a good place to start learning about this?

**Reply by euu2021** [reply]
[This page](https://docs.freeplane.org/#/scripting/Scripting) has a step by step guide teaching how to create scripts. 

If you simply want to append a # sign before the copied node ID, this script copies the node ID to the clipboard and appends a  # before it:
`textUtils.copyToClipboard('#' + node.nodeID)`

I don't use this kind of links, so I don't know the scripts and addons that exists related to it, but I remember seeing people creating a lot of things in order to make linking easier. For example, doing a quick search in the scripts collection, I found [this one](https://docs.freeplane.org/#/scripting/Scripts_collection?id=insert-link-to-a-node-in-another-map).

**Comment by macmarrum** [comment]
To create a link to another node, I would use the built-in menu items

1. `Insert->Link->Set link anchor` – to copy to clipboard the selected node's #ID
2. `Insert->Link->Make link to anchor` – to paste the copied #ID and create a link

This works also across mindmaps.
